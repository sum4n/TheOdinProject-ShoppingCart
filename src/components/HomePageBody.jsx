const HomePageBody = () => {
  return (
    <div>
      This is home page body
      <p>It will hold the page container</p>
    </div>
  );
};

export default HomePageBody;
