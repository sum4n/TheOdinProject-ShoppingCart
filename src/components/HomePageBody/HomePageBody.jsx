import styles from "./HomePageBody.module.css";
import image from "./shopping_cart.webp";
import { Link } from "react-router-dom";
import NavigationBar from "../NavigationBar/NavigationBar";

const HomePageBody = () => {
  return (
    <>
      <NavigationBar />
      <div className={styles.body}>
        <div className={styles.divText}>
          <p>Welcome to Odin Project Shopping Cart</p>
          <Link className={styles.btnShop} to="shopPage">
            Shop
          </Link>
        </div>
        <img className={styles.imgHomeBody} src={image} alt="a shopping cart" />
      </div>
    </>
  );
};

export default HomePageBody;
