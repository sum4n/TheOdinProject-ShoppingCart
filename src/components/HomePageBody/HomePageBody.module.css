.body {
  background-color: rgb(252, 175, 61);
  height: 80vh;
  margin: 8px 2% 0% 2%;
  padding: 2%;
  border-radius: 6px;

  display: flex;
  justify-content: space-around;
  align-items: center;
  gap: 14px;

  box-shadow: 0px 0px 8px rgb(63, 63, 63);
}

.divText {
  font-size: 2em;
  /* text-shadow: 0px 0px 8px rgb(63, 63, 63); */
}

button {
  border: none;
  border-radius: 4px;
  background-color: rgb(247, 119, 97);
  box-shadow: 1px 1px 1px black;
  width: 100px;
  height: 30px;
  font-weight: bold;
}

button:hover {
  box-shadow: 2px 2px 4px black;
}

.imgHomeBody {
  max-width: 80%;
  max-height: 80%;
  border-radius: 6px;
  /* border: 2px solid black; */
  box-shadow: 0px 0px 8px rgb(63, 63, 63);
}

.btnShop {
  display: block;
  text-decoration: none;
  font-size: medium;
  font-weight: bold;
  text-align: center;

  box-shadow: 1px 1px 1px black;
  border-radius: 6px;
  width: 100px;
  height: 30px;
  padding: 10px 10px 4px 10px;

  color: black;
  background-color: tomato;
}

.btnShop:hover {
  color: black;
  box-shadow: 2px 2px 4px black;
}
