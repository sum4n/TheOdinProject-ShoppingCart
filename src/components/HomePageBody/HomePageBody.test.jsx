import { render, screen } from "@testing-library/react";
import { describe, expect } from "vitest";
import HomePageBody from "./HomePageBody";
import { MemoryRouter } from "react-router-dom";

describe("HomePageBody component", () => {
  it("renders with 1 para, 1 link and 1 image", () => {
    const { container } = render(
      <MemoryRouter>
        <HomePageBody />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
