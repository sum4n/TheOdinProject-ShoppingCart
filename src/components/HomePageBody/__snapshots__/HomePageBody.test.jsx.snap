// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePageBody component > renders with 1 para, 1 link and 1 image 1`] = `
<div>
  <div
    class="_body_2c184d"
  >
    <div
      class="_divText_2c184d"
    >
      <p>
        Welcome to Odin Project Shopping Cart
      </p>
      <a
        class="_btnShop_2c184d"
        href="/shopPage"
      >
        Shop
      </a>
    </div>
    <img
      alt="a shopping cart"
      class="_imgHomeBody_2c184d"
      src="/src/components/HomePageBody/shopping_cart.webp"
    />
  </div>
</div>
`;
